src/launchpad/sentry_client.py

```diff
@@ -27,36 +49,34 @@ def __init__(self, base_url: str) -> None:
         if not self.shared_secret:
             raise RuntimeError("LAUNCHPAD_RPC_SHARED_SECRET must be provided or set as environment variable")
 
+        self.session = create_retry_session()
+
     def download_artifact(self, org: str, project: str, artifact_id: str) -> Dict[str, Any]:
         """Download preprod artifact."""
         endpoint = f"/api/0/internal/{org}/{project}/files/preprodartifacts/{artifact_id}/"
         url = self._build_url(endpoint)
 
-        try:
-            logger.debug(f"GET {url}")
-            response = requests.get(url, headers=self._get_auth_headers(), timeout=120, stream=True)
-
-            if response.status_code != 200:
-                return self._handle_error_response(response, "Download")
-
-            # Read content with size limit
-            content = b""
-            for chunk in response.iter_content(chunk_size=8192):
-                if chunk:
-                    content += chunk
-                if len(content) > 5 * 1024 * 1024 * 1024:  # 5GB limit
-                    logger.warning("Download truncated at 5GB")
-                    break
-
-            return {
-                "success": True,
-                "file_content": content,
-                "file_size_bytes": len(content),
-                "headers": dict(response.headers),
-            }
-        except Exception as e:
-            logger.error(f"Download failed: {e}")
-            return {"error": str(e)}
+        logger.debug(f"GET {url}")
+        response = self.session.get(url, headers=self._get_auth_headers(), timeout=120, stream=True)
+
+        if response.status_code != 200:
+            return self._handle_error_response(response, "Download")
+
+        # Read content with size limit
+        content = b""
+        for chunk in response.iter_content(chunk_size=8192):
+            if chunk:
+                content += chunk
+            if len(content) > 5 * 1024 * 1024 * 1024:  # 5GB limit
+                logger.warning("Download truncated at 5GB")
+                break
+
+        return {
+            "success": True,
+            "file_content": content,
+            "file_size_bytes": len(content),
+            "headers": dict(response.headers),
+        }
```

Comment:
The DX of dicts (both types and access) are quite bad for Python vs. TypeScript so what it very idiomatic in TypeScript (making the return type an anonymous object with some fields) is not the most common in Python.

If there are only ~2 fields it's more common to use a tuple:
```
def foo():
  ...
  success = True
  content = "bar"
  return success,  content
```

However if there are many fields the fastest thing is to use a `NamedTuple`:
```
class DownloadResult(NamedTuple):
    success: bool
    file_content: string
    file_size_bytes: int
```
Which is a magic way to define an object with some fields (also handles a bunch of the operators). See:

https://typing.python.org/en/latest/spec/namedtuples.html#named-tuples